src/rgw/rgw_rados.cc

```diff
@@ -2235,15 +2234,10 @@ int RGWRados::create_bucket(const RGWUserInfo& owner, rgw_bucket& bucket,
     info.owner = owner.user_id;
     info.zonegroup = zonegroup_id;
     info.placement_rule = selected_placement_rule;
-    info.layout.current_index.layout.type = rule_info.index_type;
     info.swift_ver_location = swift_ver_location;
     info.swift_versioning = (!swift_ver_location.empty());
-    if (pmaster_num_shards) {
-      info.layout.current_index.layout.normal.num_shards = *pmaster_num_shards;
-    } else {
-      info.layout.current_index.layout.normal.num_shards = bucket_index_max_shards;
-    }
-    info.layout.current_index.layout.normal.hash_type = rgw::BucketHashType::Mod;
+    init_default_bucket_layout(cct, info, svc.zone->get_zone());
+    info.layout.current_index.layout.type = rule_info.index_type;
```

Comment:
we still need to apply the `pmaster_num_shards` stuff here since it was moved out of `init_default_bucket_layout()`! eventually we won't need it anymore, so maybe add a note:
```c++
    if (pmaster_num_shards) {
      // TODO: remove this once sync doesn't require the same shard count
      info.layout.current_index.layout.normal.num_shards = *pmaster_num_shards;
    }
```

Reply:
oops right!